src/server/session.ts

```diff
@@ -2506,9 +2508,12 @@ namespace ts.server {
 
             let request: protocol.Request | undefined;
             let relevantFile: protocol.FileRequestArgs | undefined;
+            let commandSucceeded = false;

```

Comment:
I don't think you need to keep this variable, The catch block implies commandSucceedded = false while location of `perfLogger.logStopCommand("" + request.command, "Success");` implies success

Reply:
The idea behind this variable is indicated in the comment below:

```
// Note: Log before writing the response, else the editor can complete its activity before the server does
// Set 'commandSucceded' flag to ensure logStopCommand doesn't get called twice (e.g. if doOutput throws)
```

The idea being that the command could complete and we `logStopCommand` with "Success", and then `doOutput` could throw an exception and we would do a second `logStopCommand` with an error in the catch block. I don't know how likely it is for that to happen.

Reply:
Ok I removed the variable based on our conversation.